python_training/XO.py

```diff
@@ -0,0 +1,181 @@
+import random
+
+PLAYER1_SYMBOL = "X"
+PLAYER2_SYMBOL = "O"
+
+
+class XOGame:
+    """
+    Creates a XO game for two players, in any wanted board size (n>2).
+    """
+
+    def __init__(self, player1, player2, board_size):
+        """
+        Initiates the XO game.
+
+        :param player1: The name of the first player
+        :param player2: The name of the second player
+        :param board_size: The length of the boards rows and columns.
+        """
+        # Create a mapping between the player representation and name
+        self.players = {0: player1, 1: player2}
+        # Create mapping between player representation and printable symbol
+        self.symbols_mapping = {0: PLAYER1_SYMBOL, 1: PLAYER2_SYMBOL}
+        self.board_size = board_size
+        # Range is used so each row would create a different list in the memory
+        self.board = [[None] * board_size for row in range(self.board_size)]
+        self.turns_counter = 0
+        self.current_player = random.choice([*self.players])
+
+    def start_game(self):
+        """
+        Calls for turns to be taken until the game is over.
+        """
+        self.print_board()
+        while not self.has_game_ended():
+            self.take_turn()
+            self.print_board()
+
+    def print_board(self):
+        """
+        Prints the current board of the game.
+        """
+        # Prints the first line that looks like this (for n = 3):
+        #   0 1 2
+        print('   {}'.format(' '.join([str(num) for num in range(self.board_size)])))
+        # Creates a line that separates the board rows (calculated according to the number of slots)
+        # Eventually the line would look like this -------------
+        separator_line = '  {}'.format('-' * (3 * self.board_size - (self.board_size - 1)))
+        for row_number in range(self.board_size):
+            # Create a list of printable values according to our mapping.
+            printable_values = []
+            for value in self.board[row_number]:
+                if value is not None:
```

Comment:
A nice one-liner option:
`[printable_values.append(self.symbols_mapping[value] if value is not None else ' ') for value in self.board[row_number]]`